api/src/main/java/org/pmiops/workbench/api/CloudTaskUserController.java

```diff
@@ -49,73 +55,76 @@ public class CloudTaskUserController implements CloudTaskUserApiDelegate {
   private final InitialCreditsService initialCreditsService;
+  private final Provider<Stopwatch> stopwatchProvider;
+  private final UserService userService;
 
   CloudTaskUserController(
-      UserDao userDao,
-      CloudResourceManagerService cloudResourceManagerService,
-      UserService userService,
       AccessModuleService accessModuleService,
+      CloudResourceManagerService cloudResourceManagerService,
       InitialCreditsBatchUpdateService freeTierBillingUpdateService,
-      InitialCreditsService initialCreditsService) {
-    this.userDao = userDao;
-    this.cloudResourceManagerService = cloudResourceManagerService;
-    this.userService = userService;
+      InitialCreditsService initialCreditsService,
+      Provider<Stopwatch> stopwatchProvider,
+      UserService userService) {
     this.accessModuleService = accessModuleService;
+    this.cloudResourceManagerService = cloudResourceManagerService;
     this.freeTierBillingUpdateService = freeTierBillingUpdateService;
     this.initialCreditsService = initialCreditsService;
+    this.stopwatchProvider = stopwatchProvider;
+    this.userService = userService;
   }
 
   @Override
   public ResponseEntity<Void> auditProjectAccessBatch(List<Long> userIds) {
-    log.info(String.format("Audit project access for %d users...", userIds.size()));
-    int errorCount = 0;
-    for (long userId : userIds) {
-      DbUser user = userDao.findUserByUserId(userId);
-
-      // TODO(RW-2062): Move to using the gcloud api for list all resources when it is available.
-      try {
-        List<String> unauthorizedLogs =
-            cloudResourceManagerService.getAllProjectsForUser(user).stream()
-                .filter(
-                    project ->
-                        project.getParent() == null
-                            || !(ALLOWED_PARENT_IDS.contains(getId(project.getParent()))))
-                .map(
-                    project ->
-                        String.format(
-                            "%s in %s %s",
-                            project.getName(),
-                            Optional.ofNullable(project.getParent())
-                                .map(this::getType)
-                                .orElse("[type unknown]"),
-                            Optional.ofNullable(project.getParent())
-                                .map(this::getId)
-                                .orElse("[id unknown]")))
-                .collect(Collectors.toList());
-        if (!unauthorizedLogs.isEmpty()) {
-          log.warning(
-              "User "
-                  + user.getUsername()
-                  + " has access to projects: "
-                  + String.join(", ", unauthorizedLogs));
-        }
-      } catch (IOException e) {
-        log.log(Level.SEVERE, "failed to audit project access for user " + user.getUsername(), e);
-        errorCount++;
-      }
-    }
+    int errorCount = processDbUserBatch(userIds, "auditing project access", this::auditOneUser);
```

Comment:
new pattern: processDbUserBatch() and processUserIdBatch() which both take function arguments about what do to individual users.  These generic methods handle pre-run and post-run logging as well as logging elapsed time.